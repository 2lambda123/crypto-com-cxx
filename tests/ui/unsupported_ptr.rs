#[cxx::bridge]
mod ffi {
    unsafe extern "C++" {
        type C;

        fn not_unsafe_ptr(c: *mut C);
    }
}

#[cxx::bridge]
mod ffi2 {
    unsafe extern "C++" {
        type C;

        fn get_neither_const_nor_mut() -> *C;
    }
}

#[cxx::bridge]
mod ffi3 {
    unsafe extern "C++" {
        type C;

        fn get_ptr_ptr() -> *mut *mut C;
        fn get_ptr_reference() -> *mut & C;
        fn get_reference_ptr() -> & *mut C;
    }
}

fn main() {}
