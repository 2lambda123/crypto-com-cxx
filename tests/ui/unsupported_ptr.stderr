error: expected mut or const in raw pointer type
  --> $DIR/unsupported_ptr.rs:15:43
   |
15 |         fn get_neither_const_nor_mut() -> *C;
   |                                           ^ expected mut or const in raw pointer type
   |
   = help: use `*mut T` or `*const T` as appropriate

error: pointer argument requires that the function be marked unsafe
 --> $DIR/unsupported_ptr.rs:6:27
  |
6 |         fn not_unsafe_ptr(c: *mut C);
  |                           ^^^^^^^^^

error: expected `const` or `mut`
  --> $DIR/unsupported_ptr.rs:15:44
   |
15 |         fn get_neither_const_nor_mut() -> *C;
   |                                            ^

error: unsupported pointer type
  --> $DIR/unsupported_ptr.rs:24:29
   |
24 |         fn get_ptr_ptr() -> *mut *mut C;
   |                             ^^^^^^^^^^^

error: unsupported pointer type
  --> $DIR/unsupported_ptr.rs:25:35
   |
25 |         fn get_ptr_reference() -> *mut & C;
   |                                   ^^^^^^^^

error: unsupported reference type
  --> $DIR/unsupported_ptr.rs:26:35
   |
26 |         fn get_reference_ptr() -> & *mut C;
   |                                   ^^^^^^^^
